app/support/reservations/duration_change_validations.rb

```diff
@@ -0,0 +1,32 @@
+class Reservations::DurationChangeValidations
+  include ActiveModel::Validations
+
+  validate :start_time_not_changed, unless: Proc.new { |r| r.reservation.reserve_start_at_editable? }
+  validate :duration_not_shortened, if: Proc.new { |r| Time.zone.now >= r.reservation.reserve_start_at }
+
+  def initialize(reservation)
+    @reservation = reservation
+  end
+
+  def reservation
+    @reservation
+  end
+
+  def copy_errors!
+    errors.each do |e|
+      reservation.errors.add(e)
+    end
+  end
+
+  def start_time_not_changed
+    if reservation.reserve_start_at_changed?
```

Comment:
Yea, I was on the fence about this. Having the proc with the if/unless feels nice because we can scroll through the list of validations and see the preconditions for when each validation is fired.

IDK, I don't feel strongly, so if you prefer the other way I can change it.